### ISSUE TITLE
Class to class member does not work, debugging is 'aborted'

### ISSUE THREAD

**Issue by FullValueRider** [WaynePhillipsEA/twinbasic#329]
**Describe the bug**
twinBasic disappears off into hyperspace leading to an 'cancelled' message and the debug console is cleared.

**To Reproduce**
I'm trying to emulate Rubberduck unit tests for a library I'm developing.  Code is from VBA.  The Word templates containing the code work fine.  The Rubberduck tests work as expected and all tests in the test suite pass.  When I transferred the code to twinbasic the code crashes without any error messages.  All I see is a 'cancelled' message in the debug console.  The debug console is cleared straight after the 'cancelled' message is printed.

The library is attached

[VBALib.zip](https://github.com/WaynePhillipsEA/twinbasic/files/6937978/VBALib.zip)

1 Run the main function in Main.Twin
The message "Tested complete" is printed in the debug console.
2 Uncomment Test 5
3 Run the main function is Main.Twin
twinbasic pauses, eventually printing the message 'cancelled' and then the debug console is immediately cleared.

I have tried debugging but cannot get further than the line.
The Lyst Class, which is the class being tested at the time, is a wrapper for the mscorlib ArrayList.  The wrapper for the AddRange method is where the error occurs  at the line.

```
If Types.Group.IsNotIterableItemsByEnum(ipIterable) Then Exit Function
```
It appears that something related to Types.Group.IsNotIterableItemsByEnum(ipIterable) is causing the problem.

The code is currently specified to be an 'exe' as I changed the format so that I could put the test code in the same project as the library code in an attempt to follow the debug trace.

**Comment by WaynePhillipsEA** [comment]
I've identified the problem.  tB is handling predeclared objects (and all As-New types) slightly differently than VBx which is causing an infinite recursion here.  This is the code causing a problem in your Class_Initialize routine for class TypesGrouped:

```
Public Sub Class_Initialize ()
     If Me Is TypesGrouped Then

```

For predeclared objects (and As-New types), twinBASIC currently does this;

1)  checks if the predeclared object is Nothing
2)  if so, starts the construction of the object (calling Class_Initialize in that process)
3)  and then finally sets the predeclared object variable to the initialized object.  

The recursion occurs because the predeclared object is still Nothing when the Class_Initialize is executed, so the Is-check triggers another construction of the object.... and then Class_Initialize is called again... etc, etc.

I'll get this fixed tomorrow.  Thanks for reporting.

**Comment by mansellan** [comment]
So that would be a.... [Stack Overflow](https://stackoverflow.com/)?

I'll get my coat.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.3920, thanks for reporting.   Linking #54 as integrated unit testing is coming soon.
